# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1293006047 18000
#      Wed Dec 22 03:20:47 2010 -0500
# Node ID 4c335d398c8d028076312880f269cd4a9553cd53
# Parent  277d891afae2c7b66fb4dd1120752fd000e73a6b
skewness.m: make returned value have the same class as the input

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-12-22  John W. Eaton  <jwe@octave.org>
+
+	* statistics/base/skewness.m: Make returned value have the same
+	class as the input.
+
 2010-12-21  John W. Eaton  <jwe@octave.org>
 
 	* time/datevec.m: Style fixes.
 
 2010-12-20  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__fltk_print__.m: Update figure for FLTK printing.  Bug #31884.
 
diff --git a/scripts/statistics/base/skewness.m b/scripts/statistics/base/skewness.m
--- a/scripts/statistics/base/skewness.m
+++ b/scripts/statistics/base/skewness.m
@@ -69,17 +69,17 @@ function retval = skewness (x, dim)
     endif
   endif
 
   c = sz(dim);
   idx = ones (1, nd);
   idx (dim) = c;
   x = x - repmat (mean (x, dim), idx);
   sz(dim) = 1;
-  retval = zeros (sz);
+  retval = zeros (sz, class (x));
   s = std (x, [], dim);
   ind = find (s > 0);
   x = sum (x .^ 3, dim);
   retval(ind) = x(ind) ./ (c * s(ind) .^ 3);
   
 endfunction
 
 %!error skewness ();
